### PR TITLE
Special cases `Enum` and some special props to be non-final, refs #11699

### LINKED ISSUE #11699: Defining _order_ in an enum class rejected
This generates an error, even though it's valid:

```py
from enum import Enum

class E(Enum):
    _order_ = 'X Y'  # Cannot override writable attribute "_order_" with a final one
    X = 1
    Y = 2
```

According to the [docs](https://docs.python.org/3/library/enum.html#supported-sunder-names), `_order_` is a somewhat magical attribute, but it's okay to define it.

The error started happening as a side effect of #10852. 

It looks like `_order_` actually isn't a valid attribute at runtime, so a potential fix would be to remove it from typeshed. If we want to do this right, we should also remove an explicit definition from the TypeInfo, since it's not available at runtime.

@sobolevn What do you think?

This is a regression, so marking as high priority.

### PR DESCRIPTION
**Pull request by sobolevn**
My naive attempt to fix bug with some special `Enum` props which are treated as `Final`.
I just special case it and ignore them.

This looks like the simplest solution. And it fits `Enum` pretty well, because `Enum` is special in how class / instances are treated.

Closes #11699

### CODE REVIEW

File: mypy/semanal_enum.py [hunk 1]
```diff
@@ -13,6 +14,15 @@
 from mypy.semanal_shared import SemanticAnalyzerInterface
 from mypy.options import Options
 
+# Note: 'enum.EnumMeta' is deliberately excluded from this list. Classes that directly use
+# enum.EnumMeta do not necessarily automatically have the 'name' and 'value' attributes.
+ENUM_BASES: Final = frozenset((
+    'enum.Enum', 'enum.IntEnum', 'enum.Flag', 'enum.IntFlag',
+))
+ENUM_SPECIAL_PROPS: Final = frozenset((
+    'name', 'value', '_name_', '_value_', '_order_', '__order__',
```
Comment (JelleZijlstra) [inline mypy/semanal_enum.py:23]:
```suggestion
    'name', 'value', '_name_', '_value_', '_order_',
```
Only need it once :)

Comment (sobolevn) [reply mypy/semanal_enum.py:23]: But, `__order__` is a legacy version of `_order_` if I am correct. We still can use it like so:

```python
from enum import Enum

class E(Enum):
    __order__ = 'X Y Z'
    X = 1
    Y = 2
```

Снимок экрана 2021-12-11 в 18 30 47

Or does mypy uses some magic to turn `__order__` into `_order_`?

Comment (JelleZijlstra) [reply mypy/semanal_enum.py:23]: Sorry, I misread.

**Review by JukkaL — APPROVED** [review]
Thanks! It looks like `_name_` and `_value_` are not accepted at runtime, even without overriding, but this is minor.

**Pull request merged**